src/core/seat/cursor.cpp

```diff
@@ -33,7 +33,10 @@ wf::cursor_t::cursor_t(wf::seat_t *seat)
     request_set_cursor.set_callback([&] (void *data)
     {
         auto ev = static_cast<wlr_seat_pointer_request_set_cursor_event*>(data);
-        set_cursor(ev, true);
+        if (!this->hide_ref_counter)
```

Comment:
please also move this check in the function itself

Reply:
Ah, I had made the change locally already but only pushed cube.cpp and forgot cursor.cpp